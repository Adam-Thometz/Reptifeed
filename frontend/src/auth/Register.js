import React, { useState } from "react";
import { useNavigate } from 'react-router-dom';
import Alert from "../common/Alert";
import './Register.css'

const Register = ({ register }) => {
  const navigate = useNavigate();
  document.title = "Register | Reptifeed";

  const init = {
    username: '',
    password: '',
    email: ''
  }; 

  const [formData, setFormData] = useState(init);
  const [formMessages, setFormMessages] = useState([]);

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(d => ({ ...d, [name]: value }));
  };

  const handleSubmit = async e => {
    e.preventDefault();
    const res = await register(formData);
    if (res.success) {
      navigate(`/users/${res.id}`);
    } else {
      setFormMessages(res.errors)
    };
  };

  return (
    <div className="Register">
      <h1>Register for your Reptifeed account</h1>
      <form onSubmit={handleSubmit}>
        <div className="Register-form-group">
          <label className="Register-label" htmlFor="username">Username:</label>
          <input
            className="Register-input"
            type='text'
            id='username'
            name='username'
            value={formData.username}
            onChange={handleChange}
          />
        </div>
        <div className="Register-form-group">
          <label className="Register-label" htmlFor="password">Password:</label>
          <input
            className="Register-input"
            type='password'
            id='password'
            name='password'
            value={formData.password}
            onChange={handleChange}
          />
        </div>
        <div className="Register-form-group">
          <label className="Register-label" htmlFor="email">Email:</label>
          <input
            className="Register-input"
            type='email'
            id='email'
            name='email'
            value={formData.email}
            onChange={handleChange}
          />
        </div>
        <button className="Register-submit" type="submit">Register</button>
        {formMessages.length ? <Alert type="danger" messages={formMessages} /> : null }
      </form>
    </div>
  );
};

export default Register;