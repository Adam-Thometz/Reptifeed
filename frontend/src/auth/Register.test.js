import React from "react";
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Register from "./Register";

test('it renders without crashing', () => {
  render(<MemoryRouter>
    <Register />
  </MemoryRouter>);
});

test('it matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Register />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});