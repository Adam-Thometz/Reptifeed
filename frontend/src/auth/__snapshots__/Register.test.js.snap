// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Register"
  >
    <h1>
      Register for your Reptifeed account
    </h1>
    <form>
      <div
        class="Register-form-group"
      >
        <label
          class="Register-label"
          for="username"
        >
          Username:
        </label>
        <input
          class="Register-input"
          id="username"
          name="username"
          type="text"
          value=""
        />
      </div>
      <div
        class="Register-form-group"
      >
        <label
          class="Register-label"
          for="password"
        >
          Password:
        </label>
        <input
          class="Register-input"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <div
        class="Register-form-group"
      >
        <label
          class="Register-label"
          for="email"
        >
          Email:
        </label>
        <input
          class="Register-input"
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
      <button
        class="Register-submit"
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</DocumentFragment>
`;
